src/main/java/Duke.java

```diff
@@ -1,10 +1,111 @@
+                String next = sc.next();
+                if (next.equals("list")) {
+                   ui.list(store);
+                } else if (next.equals("done")) {
+                    int taskNo = sc.nextInt();
+                    if (taskNo > store.size()) {
+                        throw new DukeException("☹ OOPS!!! No such item in the list!!!");
+                    }
+                    store.get(taskNo-1).markAsDone();
+                    this.storage.write(this.tasks);
+                    System.out.println("Nice! I've marked this task as done: ");
+                    System.out.println("  " + store.get(taskNo-1).toString());
+                } else if (next.equals("bye")) {
+                    ui.bye();
+                    break;
+                } else if (next.equals("todo") || next.equals("deadline") || next.equals("event")){
+                    Task newTask;
+                    String remainder = sc.nextLine();
+                    if (next.equals("todo")) {
+                        if (remainder.trim().isEmpty()) {
+                            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                        }
+                        newTask = new Todo(remainder.trim());
+                    } else if (next.equals("deadline")) {
+                        int position = remainder.indexOf("/");
+                        if (position == -1) {
+                            throw new DukeException("☹ OOPS!!! Not a valid deadline command");
+                        }
+                        String formattedDate = Parser.getFormattedDate(remainder.substring(position+3));
+                        newTask = new Deadline(remainder.substring(0,position).trim(), formattedDate);
+                    } else {
+                        int position = remainder.indexOf("/");
+                        if (position == -1) {
+                            throw new DukeException("☹ OOPS!!! Not a valid event command");
+                        }
+                        String formattedDate = Parser.getFormattedDate(remainder.substring(position+3));
+                        newTask = new Event(remainder.substring(0,position).trim(), formattedDate);
+                    }
+                    store.add(newTask);
+                    this.storage.write(this.tasks);
+                    System.out.println("Got it. I've added this task: ");
+                    System.out.println("  " + newTask.toString());
+                    String size = Integer.toString(store.size());
+                    System.out.println("Now you have " + size + " tasks in the list.");
+                } else if (next.equals("delete")){
+                    int taskNo = sc.nextInt();
+                    if (taskNo > store.size()) {
+                        throw new DukeException("☹ OOPS!!! No such item in the list!");
+                    }
+                    System.out.println("Noted. I've removed this task: ");
```

Comment:
Similarly, for all the "Noted. I've removed this task: ", you can consider using a `removedTask` method in `Ui`.